source/controllers/ListingController.ts

```diff
@@ -0,0 +1,94 @@
+import { Listing } from '../models/Listing.js';
+import { Subsection } from '../models/Subsection.js';
+import { User } from '../models/User.js';
+import { Controller } from './BaseController.js';
+import { headerConstants } from './config.js';
+import { v4 as uuidv4 } from 'uuid';
+
+const listing = new Controller('listing', [], ['subsectionId']);
+
+listing.read = async (req, res) => {
+	const subsecId = Number(req.params.subsectionId);
+	if (!isNaN(subsecId)) {
+		const subsection = await Subsection.findByPk(subsecId, { include: [Listing] });
+		if (subsection !== null) {
+			if (res.locals.user !== null && res.locals.user !== undefined) { // for header Login&sign up or username& sign out
+				res.render('pages/main/all_listings.ejs', { subsection, subsecId, constants: headerConstants, userstatus_name: res.locals.user.username, userstatus_page: '/user/' + res.locals.user.id, signup_out_redirect: '/user/signout', signup_out_name: 'Sign Out' });
+				return;
+			} else {
+				res.render('pages/main/all_listings.ejs', { subsection, subsecId, constants: headerConstants, userstatus_name: 'Login', userstatus_page: '/user/login', signup_out_redirect: '/user/signup', signup_out_name: 'Sign Up' });
+				return;
+			}
+		}
+	}
+	// TODO: proper redirect
+	res.sendStatus(404);
+};
+
+interface ListingCreationForm {
+	listing_name: string;
+	listing_description: string;
+	startprice: number;
+	openstatus: string;
+	catid: number;
```

Comment:
`catid` is not used and should only be determined client-side to assist in picking the correct subcategory